docs/concepts.md

```diff
@@ -196,22 +196,34 @@ and an Oak Application.
 
 ## Remote Attestation
 
-Remote attestation is a core part of Project Oak. When a client connects to an
-Oak Node, the two first establish a fresh ephemeral session key, and then they
-provide assertions to each other over a channel encrypted with such key; the
-client relies on this assertion to determine whether it is connecting to a valid
-version of the Oak Runtime (see below for what constitutes a valid version). In
-particular, the attestation includes a _measurement_ (i.e. a hash) of the Oak
-Module running in the remote enclave, cryptographically bound to the session
-itself.
-
-The client may then infer additional properties about the Oak Module running on
-the remote enclave, e.g. by means of "static attestation" certificates that are
-produced as a byproduct of compiling the Oak Module source code itself on an
-enclave and having the enclave sign a statement that binds the (hash of the)
-compiled Oak Module to some high-level properties of the source code.
-
-TODO: Expand on this.
+Remote attestation is a core part of Project Oak.
+
+When a client connects to an Oak Application (currently via a gRPC Server
+pseudo-Node), the two first establish a fresh ephemeral session key, and then
+they provide assertions to each other over a channel encrypted with such key;
+the client relies on this assertion to determine whether it is connecting to a
+valid version of the Oak Runtime (see below for what constitutes a valid
+version). In particular, the attestation includes a _measurement_ (i.e. a hash)
+of the exact version of the Oak Runtime itself in the remote enclave,
+cryptographically bound to the gRPC session and signed by the hardware root of
+trust for the remote attestation (e.g. Intel in the case of SGX).
+
+This is currently provided by the
+[Enclave Key Exchange Protocol (EKEP)](https://asylo.dev/docs/concepts/ekep.html)
+as part of the [Asylo](https://github.com/google/asylo) framework.
+
+Once the client is satisfied with the remote attestation provided by the remote
+Oak Runtime instance, it may then send data over this channel, in such a way
+that the data is end-to-end encrypted between the client and that particular
+enclave instance (i.e. the encryption is terminated _within_ the enclave on the
+server side).
+
+Note that the measurement of the Oak Runtime does not include any details about
+the specific Oak Application that is running; the client only needs to be
+convinced of the fact that it is connected to a legitimate version of the Oak
+Runtime, and that this version of the Oak Runtime will correctly propagate
+labels and enforce policies, regardless of the Oak Application that it happens
+to be running (including future versions of the same or other Oak Applications).
```

Comment:
I don't believe this paragraph!

Reply:
What specifically?

Reply:
I don't believe that just relying on the integrity of the runtime is always going to be enough.  What if there's an Application that includes a formal proof of some aspect of its functionality?  Then a client will want an chain-of-custody for the Application that links back to that proof.  What if declassifying Nodes exist? Then clients will want to be able to distinguish between Applications that do and don't include declassification, based on what other functionality they provide.

Reply:
Clarified a bit, but all that is subsumed by the Oak Runtime. In particular, declassification will be performed based on labels, which may be removed by specific node instances. How a client decides what labels to apply on the data may well be driven by some out-of-band formal proof, but the Oak Runtime is not involved in such proofs. Or do you have a different view?

Reply:
The label system should hopefully give all sorts of reassurances, but that's not the same as saying that no client of an Oak Application is ever going to want to confirm that it's connecting to a particular *Application* (or version of an Application) rather than just to an Oak Runtime that might be running any arbitrary Application.

(But this isn't really a discussion to hold up this particular PR)

Reply:
While in principle that may be technically feasible, Oak is neither necessary nor sufficient for such a use case.

If, as a client, I would like to know the exact bits of an application that is running on the server, I may as well use plain SGX or whatever other basic attestation mechanism. In this use case Oak may make it easier to reason about particular aspects of the software, but it would not really offer any novel guarantees.

Let's follow up offline though.